### PR TITLE
Add catalan language

### PR DESCRIPTION
**Pull request by guillemglez**
Words from Thesaurus dictionary, Softcatalà
(GPL v2.1,  GPL v2 - https://github.com/Softcatala/catalan-dict-tools/)

Sentences from Josep Pla texts. Extracted from lletres.cat.

Flag asset from Wikimedia
https://commons.wikimedia.org/wiki/File:Flag_of_Catalonia.svg

### CODE REVIEW

File: src/app/services/language.service.ts [hunk 1]
```diff
@@ -64,6 +76,8 @@ export class LanguageService {
     switch (language) {
       case Language.DUTCH:
         return 'nl';
+      case Language.CATALAN:
+        return 'cat';
```
Comment (CasperVerswijvelt) [inline src/app/services/language.service.ts]: Use 2 letter `ISO 639-1` code

File: src/styles.scss [hunk 2]
```diff
@@ -303,6 +303,10 @@ input:checked + .slider:before {
   background-image: url(assets/icons/flags/es.svg);
 }
 
+.flag-icon-cat {
```
Comment (CasperVerswijvelt) [inline src/styles.scss]: Use 2 letter `ISO 639-1` code

Comment (CasperVerswijvelt) [inline src/styles.scss:306]: Use 2 letter `ISO 639-1` code

File: src/app/services/language.service.ts (hunk 1 above)
Comment (CasperVerswijvelt) [inline src/app/services/language.service.ts:80]: Use 2 letter `ISO 639-1` code

File: src/assets/languages/catalan/sentences.txt [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+Semblaven racons de malenconia subtil i greu de la sensibilitat empordanesa, en allò que aquesta sensibilitat pugui tenir de més contemplatiu, de més embadalit en el nostre fi, antic, somniós paisatge, en les nostres ruïnes.
+Però a mesura que aquesta música sortí de l'ou inicial i es vessà per l'àrea catalana, rebé alguna sotragada que modificà el seu esperit.
+La sardana es tornà més vibrant, més granada, d'un color més arravatat i lluent.
+Abandonà l'ombra trista dels xiprers del meu país i aparegué en altres ambients, davant d'altres paisatges, en medis més complexos i exigents.
+Aquest primer procés va de Ventura a Garreta, passant pel vell senyor Serra, de Peralada, si la memòria no m'és infidel.
+Per a comprendre aquest procés pot ésser de gran utilitat una frase que es troba en les biografies de Pep Ventura.
+En les evocacions, els biògrafs fan constar, recollint una tradició viva, que quan aquest, amb la tenora als llavis, tocava una sardana, «semblava que s'hi adormia...».
```
Comment (CasperVerswijvelt) [inline src/assets/languages/catalan/sentences.txt]: Are « and » characters typeable on catalan keyboards? If no maybe replace with regular quotation marks

**Review by CasperVerswijvelt — CHANGES_REQUESTED** [review]
Nice work! 
Due to the way the collapsable menus work in the preferences pane, the height for each group is hard-coded. Since you added a language it probably pushes off one of the toggles in the language settings. 

Height is set in `preferences.component.scss` at line 314
```
&.preferences-group-entries-language {
   max-height: 632px;
}
```

Increasing this with 31 px (height of 1 entry) should make it correct again

File: src/assets/languages/catalan/sentences.txt (hunk 3 above)
Comment (guillemglez) [reply src/assets/languages/catalan/sentences.txt]: Ha! good catch!! Yes, it should be regular quotation marks :)

**Pull request merged**